Fix WP_Query_User when it tries to sort the results

Rahmon opened the pull request:
### Description of the Change

This changes updates to use keyword fields to sort in the WP_User_Query and adds tests to ensure we are using these fields.

### Alternate Designs

### Benefits

Elasticsearch will not fail and fall back to MySQL.

### Possible Drawbacks

### Verification Process

1. Enable Users feature
2. Go to a page/post and add a paragraph block
3. Type `@` and try to search a user e.g. `@admin`
4. Check if the results were provided by Elasticsearch.

### Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my change.
- [x] All new and existing tests passed.

<!-- _NOTE: these things are not required to open a PR and can be done afterwards / while the PR is open._ -->

### Applicable Issues

Closes: #2225 

### Changelog Entry

Fixed sort for WP_User_Query

felipeelia commented on tests/php/indexables/TestUser.php:
```diff
@@ -607,11 +612,259 @@ public function testUserQueryOrderbyDisplayName() {
+
+		$users_display_name_fetched = wp_list_pluck( $user_query->results, 'display_name' );
+
+		/**
+		 * Check if Zoey is the first user because Elasticsearch sort
+		 * is case-sensitive
+		 *
+		 * Ref: https://www.elastic.co/guide/en/elasticsearch/guide/2.x/sorting-collations.html#case-insensitive-sorting
+		 *
+		 */
+		$this->assertEquals( 'Zoey', $users_display_name_fetched[0] );
```
@Rahmon it seems to be counterintuitive to rely on a case-sensitive sorting while WP's is case insensitive. Also, for fields in the posts indexable we have [a "sortable" field](https://github.com/10up/ElasticPress/blob/develop/includes/mappings/post/7-0.php#L265) that we could replicate here. Do you mind taking care of that?

Rahmon replied:
Sure. Thanks for that @felipeelia

felipeelia commented:
@Rahmon did you test it with Elasticsearch 5.6.16 (the version shipped with WP Local Docker)? I'm using the following snippet in theme's header.php:

```
new WP_User_Query(
	[
		'search'  => 'test',
		'orderby' => 'display_name',
	]
);
```

As is, I receive a 400 error saying:
> Fielddata is disabled on text fields by default. Set fielddata=true on [display_name.sortable] in order to load fielddata in memory by uninverting the inverted index. Note that this can however use significant memory. Alternatively use a keyword field instead.

If I change `wp-content/plugins/elasticpress/includes/mappings/user/initial.php` to look like this, it works:

```
'display_name'    => array(
	'type'   => 'text',
	'fields' => array(
		'raw'      => array(
			'type'                => 'keyword',
			'ignore_above' => 10922,
		),
		'sortable' => array(
			'type'         => 'keyword',
			'ignore_above' => 10922,
			'normalizer'   => 'lowerasciinormalizer',
		),
	),
),
```

Can you please check it? Thanks

Rahmon commented:
> @Rahmon did you test it with Elasticsearch 5.6.16 (the version shipped with WP Local Docker)? I'm using the following snippet in theme's header.php:
> 
> ```
> new WP_User_Query(
> 	[
> 		'search'  => 'test',
> 		'orderby' => 'display_name',
> 	]
> );
> ```
> 
> As is, I receive a 400 error saying:
> 
> > Fielddata is disabled on text fields by default. Set fielddata=true on [display_name.sortable] in order to load fielddata in memory by uninverting the inverted index. Note that this can however use significant memory. Alternatively use a keyword field instead.
> 
> If I change `wp-content/plugins/elasticpress/includes/mappings/user/initial.php` to look like this, it works:
> 
> ```
> 'display_name'    => array(
> 	'type'   => 'text',
> 	'fields' => array(
> 		'raw'      => array(
> 			'type'                => 'keyword',
> 			'ignore_above' => 10922,
> 		),
> 		'sortable' => array(
> 			'type'         => 'keyword',
> 			'ignore_above' => 10922,
> 			'normalizer'   => 'lowerasciinormalizer',
> 		),
> 	),
> ),
> ```
> 
> Can you please check it? Thanks

@felipeelia I've updated the mapping and updated some tests to check if the query uses the Elasticsearch to retrieve the users.
